docker-proxy-junit-jupiter/src/main/java/com/palantir/docker/proxy/DockerProxyExtension.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ */
+
+package com.palantir.docker.proxy;
+
+import com.palantir.docker.compose.DockerComposeExtension;
+import com.palantir.docker.compose.configuration.ProjectName;
+import com.palantir.docker.compose.execution.DockerExecutable;
+import java.io.IOException;
+import java.util.function.Function;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+public final class DockerProxyExtension extends DockerProxyManager<DockerComposeExtension.Builder>
+        implements BeforeAllCallback, AfterAllCallback {
+    /**
+     * Creates a {@link DockerProxyExtension} which will create a proxy and DNS so that tests can interface with docker
+     * containers directly.
+     *
+     * @param dockerContainerInfoCreator A {@link Function} that creates the DockerContainerInfo to use
+     * @param classToLogFor              The class using {@link DockerProxyExtension}
+     */
+    DockerProxyExtension(
+            Function<DockerExecutable, DockerContainerInfo> dockerContainerInfoCreator,
+            Class<?> classToLogFor) {
+        super(customizer -> customizer.apply(DockerComposeExtension.builder()).build(),
```

Comment:
`.retryAttempts(0)` here?

Reply:
since we're using the generic builder, it doesn't have `retryAttempts`, it's pretty sad

Reply:
Why does the generic builder matter here? In the other place you did `DCR.builder().retryAttempts(0)`. Can't you just do `DCE.builder().retryAttempts(0)` here?

Reply:
sorry, I responded to the wrong comment, there was a comment, I think it's disappeared, where you were asking why I had to put `retryAttempts` in a particular place.